module/VuFind/src/VuFind/ILS/Driver/KohaRest.php

```diff
@@ -2449,22 +2470,36 @@ protected function mapRenewalBlockReason($reason, $itype)
     }
 
     /**
-     * Return a location for a Koha item
+     * Return a location (branch or shelving) for a Koha item
      *
      * @param array $item Item
      *
      * @return string
      */
     protected function getItemLocationName($item)
     {
-        $libraryId = (!$this->useHomeLibrary && null !== $item['holding_library_id'])
-            ? $item['holding_library_id'] : $item['home_library_id'];
-        $name = $this->translateLocation($libraryId);
-        if ($name === $libraryId) {
-            $libraries = $this->getLibraries();
-            $name = isset($libraries[$libraryId])
-                ? $libraries[$libraryId]['name'] : $libraryId;
+        $locationConfig = strtolower(trim($this->config['ItemTypeRenewalBlockMappings']['Location'] ?? 'branch'));
+        switch ($locationConfig) {
+            case 'Shelving':
```

Comment:
If you `strtolower` the value, you need to change this from `Shelving` to `shelving` or it will never match. (That change was included in my earlier example, but I should have highlighted it since it's not very obvious at a glance -- sorry about that).

Reply:
thanks very much @demiankatz . Sorry about the delay, I have pushed a fix changing this case back to lower case.